client/engine/engine.go

```diff
@@ -198,10 +205,56 @@ func (e *Engine) handleMessage(message protocols.Message) (ObjectiveChangeEvent,
 		}
 
 		allCompleted.CompletedObjectives = append(allCompleted.CompletedObjectives, progressEvent.CompletedObjectives...)
+
+		relatedProgressEvent, err := e.attemptProgressForRelatedObjectives(&updatedObjective)
+		if err != nil {
+			return ObjectiveChangeEvent{}, err
+		}
+		allCompleted.CompletedObjectives = append(allCompleted.CompletedObjectives, relatedProgressEvent.CompletedObjectives...)
+
 	}
+	return allCompleted, nil
 
+}
+
+// attemptProgressForRelatedObjectives attempts to progress any objectives that may be related to the objective that was just updated
+// An objective is related when it shares a ledger channel with the provided objective.
+// This allows progress to made on other objectives that may be unblocked after processing the updatedObjective.
+func (e *Engine) attemptProgressForRelatedObjectives(updatedObjective *protocols.Objective) (ObjectiveChangeEvent, error) {
```

Comment:
We should document that we do not expect this to be the long term approach (possibly link to a notion document in a comment, or at least file an issue).